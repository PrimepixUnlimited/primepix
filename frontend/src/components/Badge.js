import React from 'react';
import {Badge} from 'react-native-elements';

const BadgeWrapper = ({badgeStyle, containerStyle, status, value}) => (
  <Badge
    badgeStyle={badgeStyle}
    status={status}
    containerStyle={containerStyle}
    value={value}
  />
);

export default BadgeWrapper;
