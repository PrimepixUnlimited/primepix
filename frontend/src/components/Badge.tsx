import React, { FC } from 'react'
import { Badge, BadgeProps } from 'react-native-elements'

interface Props extends BadgeProps {
  // custom props
}

const BadgeWrapper: FC<Props> = ({
  badgeStyle,
  containerStyle,
  status,
  value
}) => (
  <Badge
    badgeStyle={badgeStyle}
    status={status}
    containerStyle={containerStyle}
    value={value}
  />
)

export default BadgeWrapper
