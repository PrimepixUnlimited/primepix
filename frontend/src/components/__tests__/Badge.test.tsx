// components/__tests__/Hello.tsx
import 'react-native'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Badge from '../Badge'

describe('Badge snapshot', () => {
  it('renders correctly', () => {
    const badge = renderer
      .create(<Badge status="success" value={123} />)
      .toJSON()
    expect(badge).toMatchSnapshot()
  })
})
