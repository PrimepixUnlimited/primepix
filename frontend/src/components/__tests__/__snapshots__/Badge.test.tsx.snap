// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge snapshot renders correctly 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#52c41a",
        "borderColor": "#fff",
        "borderRadius": 9,
        "borderWidth": 0.5,
        "height": 18,
        "justifyContent": "center",
        "minWidth": 18,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 12,
          "paddingHorizontal": 4,
        }
      }
    >
      123
    </Text>
  </View>
</View>
`;
